crates/net/network/src/transactions/fetcher.rs

```diff
@@ -870,7 +882,7 @@ pub(super) struct TxFetchMetadata {
     /// The number of times a request attempt has been made for the hash.
     retries: u8,
     /// Peers that have announced the hash, but to which a request attempt has not yet been made.
-    fallback_peers: LruCache<PeerId>,
+    fallback_peers: LruCache<(PeerId, usize)>,
```

Comment:
could you please make a new type
```rust
struct TxSizeMetadata {
    peer_id: PeerId
    tx_encoded_len: usize,
}
```

and then implement `Hash` and `Eq` for this, so that only the `PeerId` is considered. that way if a peer announces the same hash twice with two different sizes, it's only inserted once into the cache and the second time, the first entry is replaced.

Reply:
That makes more sense, I did consider doing it initially, will implement it by later today!

Reply:
we want to keep the cache with it's nice lru properties that suit our use case here, so a `LruCache<TxSizeMetada>`.

Reply:
although we could use an `LruMap`, and add the `PeerId` as key and `PackedOption<usize>` as the value. otherwise, we would need a method `TxSizeMetadata::get_size(&self) -> Option<usize>` which returns `None` if the size is zero. as mentioned in the issue, zero is a reserved value here.

Reply:
Hi @emhane , I saw this issue was closed. Should I reopen this? I wasn't working yesterday so didn't have a chance to get back.

Thanks!

Reply:
issue was still open, pr closed though but I reopened it :)

Reply:
Thanks! Yes, my bad, PR not issue. I have the changes ready. Should I build off of main or from my current branch? Thanks!

Reply:
off of main! so rebase your changes on main, shouldn't be too many conflicts since you already had `emhane/speed-up-request-buffered-hashes` as base which has now been merged into main

Reply:
There's actually some non-trivial changes I had to work with, so I will implement the changes on main directly!

Reply:
@emhane Is PackedOption from the Cranelift crate?

Reply:
yepp that's right! since you implemented `get_size` now manually, we don't need to use that type. we can just use the `TxSizeMetadata` type directly in the fallback peers cache, so we revert to a `LruCache` from `LruMap`, since we don't need a kv map anymore, just a set. We would have used `PackedOption<usize>` for the `tx_encoded_len` otherwise. `PackedOption` essentially does the same thing: has a reserved value (in our case 0), and stores that in place of the `Option` variant `None`. this saves memory, check out https://stackoverflow.com/a/16515488. try running that program in Rust Playground with `usize` (`tx_encoded_len` type).

Reply:
Makes perfect sense, did some playing around with it. I've made a decent amount of progress with using LruCache. Just wanted to ask: Is `peer_id` strictly unique? For the `fallback_peers.removal()` type functions, the `remove()` method expects value (as defined [here](https://github.com/AbhinavMir/reth/blob/f1a102ec38443af4db072de75d29c2c5d0a97316/crates/net/network/src/cache.rs#L60-L62)), so I can "get" the txSizeMetadata object, and pass it to the remove method, but for that I just wanted to be sure that `peer_id` is unique, and then find a way to implement `get_txSizeMetadata_from_peer_id()`. Thx!

Reply:
`peer_id` is unique!